### PR TITLE
Report Postgres idle client errors to Sentry

### PR DESCRIPTION
**Pull request by nwalters512**
This should give us slightly more visibility into what's happening when servers die.

@mwest1066 I'm wondering if there's some way we can get more useful information for `idle-in-transaction timeout` errors - maybe we should have clients store the last query that was executed with them and include that when errors are reported? It wouldn't be perfect, but it'd at least let us have something to grep for.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1865,8 +1865,11 @@ if (config.startServer) {
         };
         function idleErrorHandler(err) {
           logger.error('idle client error', err);
-          // https://github.com/PrairieLearn/PrairieLearn/issues/2396
-          process.exit(1);
+          Sentry.captureException(err);
+          Sentry.close().finally(() => {
+            // https://github.com/PrairieLearn/PrairieLearn/issues/2396
```
Comment (mwest1066) [inline server.js:1870]: Looks like this issue is no longer relevant?

**Review by mwest1066 — APPROVED** [review]

**Comment by nwalters512** [comment]
@mwest1066 I believe the idle-in-transaction timeout errors are emitted per-client. In other words, they hit this:

https://github.com/PrairieLearn/PrairieLearn/blob/723b0783c1fe955134350e221891c3cc0934aa3a/prairielib/lib/sql-db.js#L152-L154

not this:

https://github.com/PrairieLearn/PrairieLearn/blob/723b0783c1fe955134350e221891c3cc0934aa3a/prairielib/lib/sql-db.js#L148-L150

So we should always know exactly which client had the error, and thus which query (assuming we store a mapping of clients to queries).

**Comment by mwest1066** [comment]
> So we should always know exactly which client had the error, and thus which query (assuming we store a mapping of clients to queries).

Oh nice! This is awesome.

**Comment by mwest1066** [comment]
Before I approve again, did you want to fix that one outdated comment?

And did you want to implement the last-query-saving as part of this? I don't think it's critical right now, but going forward it would be massively more helpful to know which query triggered this.

File: server.js (hunk 1 above)
Comment (nwalters512) [reply server.js:1870]: True! Let me remove this.

**Comment by nwalters512** [comment]
> And did you want to implement the last-query-saving as part of this? I don't think it's critical right now, but going forward it would be massively more helpful to know which query triggered this.

Not yet; I'm going to open another PR for that shortly. Want to take a final look at this one?

**Review by mwest1066 — APPROVED** [review]

**Comment by mwest1066** [comment]
> Not yet; I'm going to open another PR for that shortly. Want to take a final look at this one?

👍 approved again!

**Pull request merged**